### PR TITLE
4670 Update end date format

### PR DESCRIPTION
**Pull request by sophiemachin**

### CODE REVIEW

**Review by alistairclark — APPROVED** [review]

**Review by this-andy — APPROVED** [review]

File: api/tests/test_scripts/unit_tests/test_project_status_for_user_default.py [hunk 1]
```diff
@@ -424,9 +424,9 @@ def test_user_b_project_status(self):
                         "anonymise_url": False,
                         "task_type_name": "Systematic review",
                         "closing_date": "2018-08-17T12:00:00+00:00",
-                        "closing_date_desc": "17-August-2018, 12:00",
-                        "earliest_start_date": "2018-09-15T12:00:00+00:00",
-                        "earliest_start_date_desc": "15-September-2018, 12:00",
+                        "closing_date_desc": "17 August 2018",
+                        "earliest_start_date": "2018-09-05T12:00:00+00:00",
+                        "earliest_start_date_desc": "05 September 2018",
```
Comment (sophiemachin) [inline api/tests/test_scripts/unit_tests/test_project_status_for_user_default.py:429]: @this-andy @alistairclark I made a quick change to check dates that start with 0 (ie changed 15 to 05). Are you happy with this formatting? Or should I remove the leading 0?

Comment (this-andy) [reply api/tests/test_scripts/unit_tests/test_project_status_for_user_default.py:429]: Can you remove leading zero

Comment (sophiemachin) [reply api/tests/test_scripts/unit_tests/test_project_status_for_user_default.py]: @this-andy I've fixed this and updated the test data project tasks to test formatting the start date and end date

File: api/endpoints/project.py [hunk 2]
```diff
@@ -296,9 +296,10 @@ def process_task(task_to_process):
             task_to_process["url"] = self.calculate_task_url(task_to_process, task_id)
 
             def f(d):
-                """Convert a date string from 'YYYY-mm-dd HH:MM' format to DATE_FORMAT"""
+                """Convert a date string from 'YYYY-mm-dd HH:MM' format to
+                DATE_FORMAT and remove leading 0s"""
                 if d is not None:
-                    return dt.fromisoformat(d).strftime(DATE_FORMAT)
+                    return dt.fromisoformat(d).strftime(DATE_FORMAT).lstrip("0")
```
Comment (this-andy) [inline api/endpoints/project.py:302]: Not clear why this approach is necessary - can we not use '-d' in DATE_FORMAT?

Comment (sophiemachin) [reply api/endpoints/project.py:302]: @this-andy I actually did try this and it didn't work first time, maybe I hadn't configured it correctly. Anyway I've now implemented this.

Comment (alistairclark) [reply api/endpoints/project.py]: Should we also strip leading zeros from months?

**Pull request merged**